lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -59,13 +59,15 @@ def collection_ancestor? collections
     end
 
     def ancestor? collection
-      return true if collection == self
+      return true if (( !collection.id.nil? && !self.id.nil? ) && ( collection.id == self.id ))
+      return true if (( collection.id.nil? || self.id.nil? ) && ( collection == self ))
```

Comment:
when you say "full object comparison" all it's doing is checking that the hash code of the object are identical. This is the most efficient path if they two objects are the same. It's at least twice as slow to compare the class and the id (a string).